lib/schemes/refresh.js

```diff
@@ -0,0 +1,328 @@
+          config.headers[this.options.tokenName] = this.$auth.getToken(this.name)
+          failedRequestQueue.push(this.$auth.ctx.app.$axios.request(config))
+        })
+
+        // Clear global queue
+        this.failedRequestQueue = []
+
+        // Retry all failed requests
+        await Promise.all(failedRequestQueue)
+      }
+
+      // End the refresh
+      this.isRefreshing = false
+    }
+  }
+
+  async _tokenRefresh (self) {
+    // Refresh endpoint is disabled
+    if (!this.options.endpoints.refresh) return
+
+    // Token is required but not available
+    if (this.options.tokenRequired && !this.$auth.getToken(this.name)) return
+
+    // Token is already being refreshed
+    if (this.isRefreshing) return
+
+    // Start the refresh
+    this.isRefreshing = true
+
+    const payloadProperties = this.options.endpoints.refresh.payloadProperties
+    const endpoint = {
+      data: {
+        [payloadProperties.refreshToken]: this._getRefreshToken()
+      }
+    }
+
+    // Only add client id to payload if enabled
+    if (payloadProperties.clientId !== false) {
+      endpoint.data[payloadProperties.clientId] = this.$auth.getClientId(this.name)
+    }
+
+    // Only add grant type to payload if enabled
+    if (payloadProperties.grantType !== false) {
+      endpoint.data[payloadProperties.grantType] = this.options.endpoints.refresh.grantType
+    }
+
+    // Try to fetch user and then set
+    return this.$auth.requestWith(
+      this.name,
+      endpoint,
+      this.options.endpoints.refresh
+    ).then(response => {
+      this._updateTokens(response)
+    }).catch(err => {
+      this._logoutLocally()
+    })
+  }
+
+  _scheduleTokenRefresh () {
+    if (this.options.endpoints.refresh.disableAutoRefresh) {
+      return
+    }
+
+    let self = this
+    let intervalDuration = (self._getExpiration() - Date.now()) * 0.75
+    if (intervalDuration < 1000) {
+      // in case you misconfigured refreshing this will save your auth-server from a self-induced DDoS-Attack
+      intervalDuration = 1000
+    }
+
+    this.refreshInterval = setInterval(function () {
+      self._tokenRefresh()
+    }, intervalDuration)
+  }
+
+  _initializeErrorInterceptor () {
+    this.$auth.ctx.app.$axios.onError(error => {
+      // If error has config and response and error status is 401
+      if (error.config && error.response && error.response.status === 401) {
+        var token = this.$auth.getToken(this.name)
+        var headerToken = error.config.headers[this.options.tokenName]
+
+        // If current token is equal to header token
+        if (token && headerToken && headerToken === token) {
+          // Add the failed request config to failedRequestQueue
+          if (!error.config.__isRetryRequest) {
+            error.config.__isRetryRequest = true
+            this.failedRequestQueue.push(error.config)
+          }
+
+          // If is not refreshing the token
+          if (!this.isRefreshing) {
+            var self = this
```

Comment:
Why `self` is used? Using an arrow function we can use `this`

Reply:
https://github.com/JoaoPedroAS51/auth-module/pull/8/commits/d91078ad98c7cb94cabceb211ba7591dd3688b5f